grommet/grommet
# Improve TextInput keyboard accessibility

#### What does this PR do?
This PR allows the user to navigate the TextInput suggestions with keyboard and keeps the active suggestion in view by scrolling through the options as the down/up arrow is pressed. The portion of the functions that check if the suggestion is before/after scroll were leveraged from grommet Select.

#### Where should the reviewer start?
src/js/components/TextInput/TextInput.js

#### What testing has been done on this PR?
This has been tested in Chrome, Safari, and FireFox using the "Suggestions" storybook example. As you use down arrow key or up arrow key to navigate through the results, the drop should scroll to keep the active suggestion in view. Also, it supports a trade off between mouse/keyboard navigation where when the mouse hovers over an option it makes that the new active index and keyboard navigation will be up from there.

A note: I wanted to add in a test to check for this, but ran into an issue with being able to check what is truly "visible" in the browser as opposed to just present in the DOM. Here is a link to others with a similar issue: https://github.com/testing-library/dom-testing-library/issues/196. This PR includes a fix, but not an added test.

#### How should this be manually tested?
Using TextInput Suggestions story book, use arrow keys to navigate through the suggestions and make sure the drop scroll with you.

#### Any background context you want to provide?
As users used arrow keys to navigate down or up in the suggestions, the drop wouldn't scroll so the active suggestion could fall out of view.

#### What are the relevant issues?
N/A.

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.
